src/main/java/org/broadinstitute/hellbender/tools/sv/cluster/CanonicalSVLinkage.java

```diff
@@ -222,12 +274,31 @@ private static int getLength(final SVCallRecord record, final int missingInserti
             return record.getLength() == null ? missingInsertionLength : Math.max(record.getLength(), 1);
         } else if (record.getType() == GATKSVVCFConstants.StructuralVariantAnnotationType.BND) {
             return record.getPositionB() - record.getPositionA() + 1;
+        } else if (record.getType() == GATKSVVCFConstants.StructuralVariantAnnotationType.CPX) {
+            return getLengthComplex(record, missingInsertionLength);
         } else {
             // TODO lengths less than 1 shouldn't be valid
             return Math.max(record.getLength() == null ? 1 : record.getLength(), 1);
         }
     }
 
+    /**
+     * Returns length to be used for complex variant matching
+     */
+    private static int getLengthComplex(final SVCallRecord record, final int missingInsertionLength) {
+        if (!record.isIntrachromosomal() || record.getPositionA() == record.getPositionB()) {
```

Comment:
I think the changes you made to the way CPX events are loaded in means that they are never interchromosomal by this check. Also, I am not sure that this will reliably identify insertion-type events, because if there is a DEL at the sink then POS != END - is that the desired behavior?

Reply:
Removed this method